test/puppetlabs/trapperkeeper/testutils/logging.clj

```diff
@@ -1,28 +1,321 @@
+(defmacro with-test-logging-debug
+  "Creates an environment for the use of the logged? test method, and
+  arranges for every event map logged within that environment to be
+  printed to *err*."
+  [& body]
+  `(let [destination# (atom [])]
+     (binding [*test-log-events* destination#]
+       (with-log-level root-logger-name :trace
+         (with-log-event-listener root-logger-name
+           (fn [event#]
+             (binding [*out* *err*]
+               (println "** Log entry:" (pr-str (event->map event#))))
+             (swap! destination# conj event#))
+           ~@body)))))
+
+(s/defn ^{:always-validate true} logged?
+  ([msg-or-pred] (logged? msg-or-pred nil))
+  ([msg-or-pred :- (s/conditional ifn? (s/pred ifn?)
+                                  string? s/Str
+                                  :else Pattern)
+    maybe-level :- (s/maybe (s/pred #(levels %)))]
+   (let [match? (cond (ifn? msg-or-pred) msg-or-pred
+                      (string? msg-or-pred) #(= msg-or-pred (:message %))
+                      :else #(re-find msg-or-pred (:message %)))
+         one-element? #(and (seq %) (empty? (rest %)))]
+     (one-element? (filter match? (map event->map @*test-log-events*))))))
+
+(defmethod clojure.test/assert-expr 'logged? [is-msg form]
+  "Asserts that exactly one event in *test-log-events* has a message
+  that matches msg-or-pred.  The match is performed via = if
+  msg-or-pred is a string, via re-find if msg-or-pred is a pattern, or
+  via (msg-or-pred event-map) if msg-or-pred is a function.  If level
+  is specified, the message's keyword level (:info, :error, etc.) must
+  also match.  For example:
+    (with-test-logging (log/info \"123\") (is (logged? #\"2\")))."
+  (assert (#{2 3} (count form)))
+  (let [[_ msg-or-pred level] form]
+    `(let [events# @@#'puppetlabs.trapperkeeper.testutils.logging/*test-log-events*]
+       (if-not (logged? ~msg-or-pred ~level)
+         (clojure.test/do-report
+          {:type :fail
+           :message ~is-msg
+           :expected '~form
+           :actual (list '~'logged events#)})
+         (clojure.test/do-report
+          {:type :pass
+           :message ~is-msg
+           :expected '~form
+           :actual (list '~'logged events#)})))))
+
+(defn reset-logging-config-after-test
+  "Fixture that will reset the logging configuration after each
+  test.  Useful for tests that manipulate the logging configuration,
+  in order to ensure that they don't affect test logging for subsequent
+  tests."
+  [f]
+  (f)
+  (pl-log/reset-logging))
+
+
+;;; Deprecated API (clojure.tools.logging specific, etc.)
 
 (def ^{:doc "A dynamic var that is bound to an atom containing all of the log entries
              that have occurred during a test, when using `with-test-logging`."
-       :dynamic true}
+       :dynamic true
+       :deprecated "1.1.2"}
   *test-logs*
   nil)
 
-(def legal-levels #{nil :trace :debug :info :warn :error :fatal})
+(def ^{:deprecated "1.1.2"} legal-levels
```

Comment:
This metadata is used quite a bit in [clojure.core](https://github.com/clojure/clojure/blob/bdc752a7fefff5e63e0847836ae5e6d95f971c37/src/clj/clojure/core.clj#L4924). There is nothing special in the language, but autodoc does handle it [correctly](https://clojure.github.io/clojure/clojure.core-api.html#clojure.core/add-classpath).  I've not used [deprecated](https://github.com/alexander-yakushev/defprecated) but that does add more rigor along with warnings.

Seems like deprecating it and removing it on the next major is the right approach here. We publish and encourage the usage of test-jars, seems like it should have the same semver properties.

I wouldn't be a fan of removing it without a semver bump. Seems like it should follow the rest